stringsubstitutor/src/main/java/org/jdbi/v3/stringsubstitutor/StringSubstitutorTemplateEngine.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jdbi.v3.stringsubstitutor;
+
+import org.apache.commons.text.StringSubstitutor;
+import org.jdbi.v3.core.statement.StatementContext;
+import org.jdbi.v3.core.statement.TemplateEngine;
+
+/**
+ * Register an instance of this interface ({@link org.jdbi.v3.core.statement.SqlStatements#setTemplateEngine}) to use an Apache Commons Text {@link StringSubstitutor} as a {@link TemplateEngine}. This lets you use any pair of strings as variable delimiters, enabling the use of syntax like <pre>select * from ${foo}</pre>, <pre>select * from &lt;foo&gt;</pre>, <pre>select * from %foo%</pre>, etc.
+ */
+@FunctionalInterface
+public interface StringSubstitutorTemplateEngine extends TemplateEngine {
```

Comment:
I'm still hung up on this being an interface. @stevenschlansker am I being too picky?

Reply:
Yeah, this doesn't feel like an interface to me either.  How about taking a `Consumer<StringSubstitutor>` as a constructor argument instead?  Then you decouple the desire for lambdas from the choice of `class` vs `interface` for this piece of api.